### PR TITLE
Throw on invalid or unregistered schema

### PR DESCRIPTION
**Pull request by carragom**

### CODE REVIEW

File: src/validator.ts [hunk 1]
```diff
@@ -30,7 +30,13 @@ export class Validator {
     } else {
       validate = ajv.compile(schema);
     }
-    if (validate === undefined) return null;
+    if (validate === undefined) {
+      if (typeof schema === 'string') {
+        throw new Error(`Unregistered schema requested: '${schema}'`);
+      } else {
+        throw new TypeError(`Error compiling schema`);
+      }
```
Comment (kraih) [inline src/validator.ts:38]: This line is not covered by tests.

**Review by kraih — CHANGES_REQUESTED** [review]

**Comment by kraih** [comment]
There is also a context method `schema` that no longer returns `null`.

File: src/validator.ts (hunk 1 above)
Comment (kraih) [reply src/validator.ts:38]: Also unnecessary use of a template literal.

Comment (carragom) [reply src/validator.ts]: This should both be fixed on the next commit

**Comment by kraih** [comment]
Please squash your commits.

File: src/validator.ts [hunk 2]
```diff
@@ -30,7 +30,9 @@ export class Validator {
     } else {
       validate = ajv.compile(schema);
     }
-    if (validate === undefined) return null;
+    if (validate === undefined) {
+      throw new Error(`Invalid schema: '${schema}'`);
```
Comment (kraih) [inline src/validator.ts:34]: That doesn't seem very helpful if the schema is not a string.

Comment (kraih) [reply src/validator.ts:34]: Also no need to use a block.

Comment (kraih) [reply src/validator.ts:34]: Looks like `ajv` throws for actual schema errors, so the message is probably fine.

**Review by kraih — APPROVED** [review]

**Pull request merged**